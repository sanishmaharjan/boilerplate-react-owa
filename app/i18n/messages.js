export default {
    en: {
        LOGOUT: 'Logout',
        CHOOSE_LANGUAGE: 'Choose language',
        SUCCESS: 'Success',
        ERROR: 'Error',
        INFO: 'Info',
        HOME_PAGE: 'Home page',
        BLINK_PAGE: 'Blink page',
        USER_INFORMATION: 'User Information',
        USER_UUID: 'User Uuid',
        NAME: 'Name',
        USER_NAME: 'User Name',
        NOTIFICATION: 'Notification',
        MESSAGE: 'Message'
    },
    es: {
        LOGOUT: 'Cerrar sesión',
        CHOOSE_LANGUAGE: 'Elige lengua',
        SUCCESS: 'Éxito',
        ERROR: 'Error',
        INFO: 'Información',
        HOME_PAGE: 'Página de inicio',
        BLINK_PAGE: 'Página de parpadeo',
        USER_INFORMATION: 'informacion del usuario',
        USER_UUID: 'Usuario Uuid',
        NAME: 'Nombre',
        USER_NAME: 'Nombre de usuario',
        NOTIFICATION: 'Notificación',
        MESSAGE: 'Mensaje'
    },
    fr: {
        LOGOUT: 'Connectez - Out',
        CHOOSE_LANGUAGE: 'Choisissez la langue',
        SUCCESS: 'Succès',
        ERROR: 'Erreur',
        INFO: 'Info',
        HOME_PAGE: 'Page d\'accueil',
        BLINK_PAGE: 'Blink page',
        USER_INFORMATION: 'informations de l\'utilisateur',
        USER_UUID: 'Utilisateur Uuid',
        NAME: 'prénom',
        USER_NAME: 'Nom d\'utilisateur',
        NOTIFICATION: 'Notification',
        MESSAGE: 'Message'
    },
    pt: {
        LOGOUT: 'Sair',
        CHOOSE_LANGUAGE: 'Escolha o seu idioma',
        SUCCESS: 'Sucesso',
        ERROR: 'Erro',
        INFO: 'Info',
        HOME_PAGE: 'Pagina inicial',
        BLINK_PAGE: 'Página de piscar',
        USER_INFORMATION: 'Informação do usuário',
        USER_UUID: 'Usuário Uuid',
        NAME: 'Nome',
        USER_NAME: 'Nome de usuário',
        NOTIFICATION: 'Notificação',
        MESSAGE: 'mensagem'
    }
};
