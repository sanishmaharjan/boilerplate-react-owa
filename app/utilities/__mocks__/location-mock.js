Object.defineProperties(window.location, {
    host: {
        writable: true,
        value: '192.168.33.10'
    },
    hostname: {
        writable: true,
        value: '192.168.33.10'
    },
    href: {
        writable: true,
        value: 'https://192.168.33.10/openmrs/owa/boilerplate/index.html'
    },
    origin: {
        writable: true,
        value: 'https://192.168.33.10'
    },
    pathname: {
        writable: true,
        value: '/openmrs/owa/boilerplate/index.html'
    },
    port: {
        writable: true,
        value: ''
    },
    protocol: {
        writable: true,
        value: 'https:'
    }
});
