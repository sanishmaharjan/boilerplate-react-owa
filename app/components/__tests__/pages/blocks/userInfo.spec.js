import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {IntlProvider} from 'react-intl';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import UserInfo from 'components/pages/blocks/userInfo';
import messages from 'i18n/messages';

require('utilities/__mocks__/location-mock');
const intlProvider = new IntlProvider({locale: 'en', messages: messages['en']}, {});
const {intl} = intlProvider.getChildContext();
const testData = {
    context: {
        intl: intl
    },
    sleep: (milisec) => {
        return new Promise((resolve, reject) => {
            setTimeout(() => {
                return resolve(true);
            }, milisec);
        });
    }
};

describe('UserInfo',  () => {
    beforeAll(() => {
        let mock = new MockAdapter(axios);
        const data = {
            sessionId: '0680FE18902BDA7BBCB4AF824747D6FB',
            authenticated: true,
            user: {
                uuid: 'c1c21e11-3f10-11e4-adec-0800271c1b75',
                username: 'admin',
                person: {
                    display: 'Admin User',
                    uuid: 'c1bc22a5-3f10-11e4-adec-0800271c1b75'
                }
            },
            locale: 'en',
            allowedLocales: ['en', 'es', 'fr', 'it', 'pt_BR']
        };

        mock.onGet('https://192.168.33.10/openmrs/ws/rest/v1/session').reply(200, data);
    });

    it('renders correctly', async () => {
        const userInfo = shallow(<UserInfo />, {context: testData.context});

        await testData.sleep(100);
        userInfo.update();

        expect(userInfo.find('#user-uuid').text().trim()).toBe('c1c21e11-3f10-11e4-adec-0800271c1b75');
        expect(userInfo.find('#user-name').text().trim()).toBe('admin');
        expect(shallowToJson(userInfo)).toMatchSnapshot();
    });
});
