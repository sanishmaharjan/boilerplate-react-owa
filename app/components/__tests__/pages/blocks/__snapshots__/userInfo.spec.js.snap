// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfo renders correctly 1`] = `
<fieldset
  className="block-fieldset"
>
  <legend>
      
    User Information
      
  </legend>
  <p>
    <label>
      User Uuid
      :
    </label>
    <span
      id="user-uuid"
    >
       
      c1c21e11-3f10-11e4-adec-0800271c1b75
       
    </span>
  </p>
  <p>
    <label>
      Name
      :
    </label>
    <span>
       
      Admin User
       
    </span>
  </p>
  <p>
    <label>
      User Name
      :
    </label>
    <span
      id="user-name"
    >
       
      admin
       
    </span>
  </p>
</fieldset>
`;
